### PR TITLE
No Content-Type text/html on empty exception views [master]

### PR DESCRIPTION
**Pull request by mauritsvanrees**
Fixes issue #1089 for Zope 5.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -11,6 +11,8 @@ https://github.com/zopefoundation/Zope/blob/4.x/CHANGES.rst
 5.8 (unreleased)
 ----------------
 
+- Only set response header Content-Type as text/html on exception views when the response has content.
```
Comment (dataflake) [inline CHANGES.rst:14]:
```suggestion
- Only set response header Content-Type as text/html on exception views when
  the response has content.
  (`#1089 <https://github.com/zopefoundation/Zope/issues/1089>`_)
```

File: src/ZPublisher/WSGIPublisher.py [hunk 2]
```diff
@@ -149,13 +149,22 @@ def _exc_view_created_response(exc, request, response):
...
+        body = view()
+
         # Explicitly set the content type header if it's not there yet so
-        # the response doesn't get served with the text/plain default
-        if not response.getHeader('Content-Type'):
+        # the response does not get served with the text/plain default.
+        # But only do this when there is a body.
+        # An empty body may indicate a 304 NotModified from Plone,
+        # and setting a content type header will change the stored header
+        # in for example Varnish.
```
Comment (dataflake) [inline src/ZPublisher/WSGIPublisher.py:160]: This is not Plone or Varnish-specific.

```suggestion
        # An empty body may indicate a 304 NotModified response,
        # and setting a content type header will change the stored header
        # in caching servers such as Varnish.
```

File: src/ZPublisher/WSGIPublisher.py [hunk 3]
```diff
@@ -149,13 +149,22 @@ def _exc_view_created_response(exc, request, response):
...
+        # See https://github.com/zopefoundation/Zope/pull/1088
```
Comment (dataflake) [inline src/ZPublisher/WSGIPublisher.py:161]: Refer to the issue and not the PR

```suggestion
        # See https://github.com/zopefoundation/Zope/issues/1089
```

**Review by dataflake — CHANGES_REQUESTED** [review]

**Comment by mauritsvanrees** [comment]
Merged. Thanks for the review!

**Pull request merged**